core/store/config/general_config.go

```diff
@@ -946,47 +937,17 @@ func (c *generalConfig) P2PPeerstoreWriteInterval() time.Duration {
 }
 
 // P2PPeerID is the default peer ID that will be used, if not overridden
-func (c *generalConfig) P2PPeerID() (p2pkey.PeerID, error) {
-	var err error
-	func() {
-		c.p2ppeerIDmtx.Lock()
-		defer c.p2ppeerIDmtx.Unlock()
-		if c.viper.GetString(EnvVarName("P2PPeerID")) == "" {
-			if c.ks == nil {
-				err = errors.New("keystore was not set on config")
-				return
-			}
-			keys, err2 := c.ks.P2P().GetAll()
-			if err2 != nil {
-				logger.Warnw("Failed to load keys, falling back to env", "err", err2.Error())
-				return
-			}
-			if len(keys) > 0 {
-				peerID := keys[0].PeerID()
-				logger.Debugw("P2P_PEER_ID was not set, using the first available key", "peerID", peerID.String())
-				c.viper.Set(EnvVarName("P2PPeerID"), peerID)
-				if len(keys) > 1 {
-					logger.Warnf("Found more than one P2P key in the database, but no P2P_PEER_ID was specified. Defaulting to first key: %s. Please consider setting P2P_PEER_ID explicitly.", peerID.String())
-				}
-			} else {
-				err = errors.New("the configuration variable P2P_PEER_ID was not specified and found no P2P keys in the database. Please set P2P_PEER_ID explicitly or add at least one key")
-			}
-		}
-	}()
-	if err != nil {
-		return "", err
-	}
-
+func (c *generalConfig) P2PPeerID() p2pkey.PeerID {
+	var pid p2pkey.PeerID
 	pidStr := c.viper.GetString(EnvVarName("P2PPeerID"))
 	if pidStr != "" {
-		var pid p2pkey.PeerID
 		err := pid.UnmarshalText([]byte(pidStr))
 		if err != nil {
-			return "", errors.Wrapf(ErrInvalid, "P2P_PEER_ID is invalid %v", err)
+			logger.Error(errors.Wrapf(ErrInvalid, "P2P_PEER_ID is invalid %v", err))
 		}
-		return pid, nil
+		return p2pkey.PeerID("")
```

Comment:
Should this one be up inside the `err != nil` block?
What about reformatting for less indentation:
```go
	pidStr := c.viper.GetString(EnvVarName("P2PPeerID"))
	if pidStr == "" {
		return ""
	}
	var pid p2pkey.PeerID
	if err := pid.UnmarshalText([]byte(pidStr)); err != nil {
		logger.Error(errors.Wrapf(ErrInvalid, "P2P_PEER_ID is invalid %v", err))
		return ""
	}
	return pid
```

Reply:
good rewrite :+1:

Reply:
I picked up this habit from an early version of cockroachdb's style guide: https://wiki.crdb.io/wiki/spaces/CRDB/pages/181371303/Go+Golang+coding+guidelines#Reduce-Nesting
Lots of good readability stuff in there.